src/crypto/crypto_openssl.c

```diff
@@ -1218,6 +1218,43 @@ int crypto_get_random(void *buf, size_t len)
 int omac1_aes_vector(const u8 *key, size_t key_len, size_t num_elem,
 		     const u8 *addr[], const size_t *len, u8 *mac)
 {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+	EVP_MAC_CTX *ctx = NULL;
+	EVP_MAC *emac;
+	int ret = -1;
+	size_t outlen, i;
+	OSSL_PARAM params[2];
+	char *cipher = NULL;
+	if (TEST_FAIL())
+		return -1;
+	emac = EVP_MAC_fetch(NULL, "CMAC", NULL);
+
+	if (key_len == 32)
+		cipher = "aes-256-cbc";
+	else if (key_len == 24)
+		cipher = "aes-192-cbc";
+	else if (key_len == 16)
+		cipher = "aes-128-cbc";
+
+	params[0] = OSSL_PARAM_construct_utf8_string("cipher", cipher, 0);
+	params[1] = OSSL_PARAM_construct_end();
+
+	if (!emac || !cipher ||
+	    !(ctx = EVP_MAC_CTX_new(emac)) ||
+	    EVP_MAC_init(ctx, key, key_len, params) != 1)
+		goto fail;
+
+	for (i = 0; i < num_elem; i++) {
+		if (!EVP_MAC_update(ctx, addr[i], len[i]))
+			goto fail;
+	}
+	if (EVP_MAC_final(ctx, mac, &outlen, 16) != 1 || outlen != 16)
+		goto fail;
+	ret = 0;
+fail:
+	EVP_MAC_CTX_free(ctx);
```

Comment:
are we missing `EVP_MAC_free(mac);` here?

reference: Example in https://www.openssl.org/docs/man3.1/man3/EVP_MAC_init.htm

Reply:
@wumiaont , can you check this comment?

Reply:
Yes. That does seem to be missing. I will double check and verify that fix. Thanks for finding this out.

Reply:
The mac parameter is an input variable, it may be a little different with the sample code. Is the caller to free it?

aes_siv_encrypt --> aes_s2v --> omac1_aes_vector
See https://github.com/sonic-net/sonic-wpa-supplicant/blob/413704a6ccef8321667712c518e595878ff02251/src/crypto/aes-siv.c#L127C2-L127C23

@r12f , @wumiaont , please double check if necessary.

If needed, do we need to add it in line1288?

Reply:
@xumia , maybe I am reading the wrong code... this is what I am seeing. The mac is allocated using EVP_MAC_FETCH

![image](https://github.com/sonic-net/sonic-wpa-supplicant/assets/1533278/1a0440e1-35b2-4739-a04a-21f4638685f9)